pyomo/core/tests/unit/test_block.py

```diff
@@ -186,7 +187,8 @@ def generator_test(self, ctype):
                 generator = list(block.component_objects(ctype, active=True, descend_into=False))
             except:
                 if issubclass(ctype, Component):
-                    self.fail("component_objects(active=True) failed with ctype %s" % ctype)
+                    print("component_objects(active=True) failed with ctype %s" % ctype)
+                    raise
```

Comment:
Why did you change these from `self.fail` statements to `print` and `raise`?

Reply:
The test started failing, but because it was suppressing the exception, I could not diagnose the error.  This way, if the test fails you still get the exception stack trace and can resolve the problem.